File: python/dp-e02.py
from sys import argv

def simple_interest(principal, rate, periods):
    return principal * (rate / 100) * periods

def compound_interest(principal, rate, periods):
    return principal * (1 + (rate / 100))**periods

if __name__ == '__main__':
    if len(argv) != 5:
        print("Usage: TYPE PRINCIPAL RATE PERIODS")

    if argv[1] == 'simple':
        print(simple_interest(*argv[2:]))
    elif argv[1] == 'compound':
        print(compound_interest(*argv[2:]))
    else:
        print('')

File: old/javascript/dp-e02.js
let args = process.argv.slice(2);
let variables = args.slice(1).map((x) => parseInt(x));


let simple = (p, r, pe) => p + (p * (r/100) * pe);
let compound = (p, r, pe) => p * Math.pow((1 + r/100), pe);

if (args.length != 4) {
  console.log("ERROR: in format TYPE PRINCIPAL RATE PERIODS");
} else if (args[0] == 'simple') {
  console.log(simple(...variables));
} else if (args[0] == 'compound') {
  console.log(compound(...variables));
} else {
  console.log("Must use either 'simple' or 'compund' for type.");
}
